resources/dnf_pkg.sh

```diff
@@ -0,0 +1,149 @@
+# == Name
+#
+# dnf.pkg
+#
+# === Description
+#
+# Manage packages via dnf.
+#
+# === Parameters
+#
+# * state: The state of the resource. Required. Default: present.
+# * package: The name of the package, or the path to the rpm-file. Required.
+# * version: The version of the package. Leave empty for first version found. Set to "latest" to always update.
+#
+# === Example
+#
+# ```bash
+# dnf.pkg --package tmux --version latest
+# dnf.pkg --package http://download1.rpmfusion.org/free/fedora/rpmfusion-free-release-$(rpm -E %fedora).noarch.rpm
+# ```
+#
+
+dnf.pkg() {
+  # Declare the resource
+  waffles_resource="dnf.pkg"
+
+  # Use dnf if available otherwise use yum
+  local _dnf_cmd="dnf"
+  if ! waffles.command_exists ${_dnf_cmd} ; then
+    _dnf_cmd="yum"
+  fi
+
+  # Check if all dependencies are installed
+  local _wrd=("${_dnf_cmd}" "rev" "cut" "tr" "tail")
```

Comment:
Since coreutils is an implicit requirement, `cut`, `tr`, and `tail` can be excluded here. On Ubuntu, `rev` is part of a `util-linux` package... let's keep `rev` here for now.